staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config_test.go

```diff
@@ -213,8 +213,8 @@ resources:
 `
 
 	incorrectConfigNoEndpointForKMS = `
-kind: EncryptionConfig
-apiVersion: v1
+kind: EncryptionConfiguration
+apiVersion: apiserver.config.k8s.io/v1
```

Comment:
can we still read the old "v1" config files? Do we have to? Do we have tests for that?

Reply:
As per https://github.com/kubernetes/kubernetes/pull/63032#discussion_r185873533, we cannot.

Reply:
Is this still open? Compare the discussion over there. No reason not to read the old apigroup files as well.

Reply:
This PR is a successor to the other one, it's the one I can manage and change. If you're asking more about the question of reading old config files, I consider that one still open, too.

I suspect we may go the way you propose. I am not a big fan of the duality it introduces, but given the fact it's already documented that way and might possibly have its users, there's very little choice.

If I read it right, we just need to add this type to the call of `scheme.AddKnownTypes()` somewhere in its legacy api group.

I wonder why the name was changed from `EncryptionConfig` to `EncryptionConfiguration`. It might have been for the single fact of underlining their incompatibility. @php-coder, do you possibly remember? We'd have to stick with the old one IIUTC.

Reply:
> I wonder why the name was changed from EncryptionConfig to EncryptionConfiguration

To be consistent with other `*Configuration` objects.

Reply:
Ah, of course, thank you!